### DISCUSSION TITLE
how the export works?

### DISCUSSION THREAD

**Discussion by CypherPotato** [pawelsalawa/sqlitestudio#4927] [Q&A] [answered]
I needed to deal with huge tables, with over 20 GB of data, and work on exporting certain columns with queries on them.

For this, the ideal is that, as it reads the lines, it exports the contents of the table to my file, however, I tried to export to CSV, got it exporting, and even after some time, no lines were written to my file.

After canceling the operation, I saw that this was written in the console:

    [15:25:01] An error occured while executing the count(*) query, thus data paging will be disabled. Error details from the database: interrupted

Why was it counting the lines before exporting? Is it to show progress or something else?

If so, there should be a way to disable this count, because it takes time in very large tables.

**Comment by pawelsalawa** [comment, marked as answer]
If I remember correctly - yes, this is the reason of counting. And your suggestion makes a lot of sense. Please report the issue.

**Reply by CypherPotato** [reply]
Got it on https://github.com/pawelsalawa/sqlitestudio/issues/4928. Thank you.
